cli/client.go

```diff
@@ -403,14 +404,58 @@ func VersionAction(c *cli.Context) error {
 	return nil
 }
 
-func checkBaseURL(c *cli.Context) (*url.URL, []rpc.DialOption, error) {
+var defaultBaseURL = "https://app.viam.com:443"
+
+func checkBaseURL(c *cli.Context, auth *config) (*url.URL, []rpc.DialOption, error) {
+	// If base URL was not specified, assume cached base URL. If no base URL is
+	// cached, assume default base URL.
 	baseURL := c.String(baseURLFlag)
+	if baseURL == "" {
+		baseURL = defaultBaseURL
+		if auth.Auth != nil && auth.Auth.BaseURL != "" {
+			baseURL = auth.Auth.BaseURL
+		}
+	}
+
+	if auth.Auth != nil && auth.Auth.BaseURL != baseURL {
+		return nil, nil, fmt.Errorf("cached base URL for this session is %q, "+
+			"please logout and login again to use provided base URL %q", auth.Auth.BaseURL, baseURL)
+	}
+
 	baseURLParsed, err := url.Parse(baseURL)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	if baseURLParsed.Scheme == "https" {
+	// Go URL parsing can place the host in Path if no scheme is provided; place
+	// Path in Host in this case.
+	if baseURLParsed.Host == "" && baseURLParsed.Path != "" {
+		baseURLParsed.Host = baseURLParsed.Path
+		baseURLParsed.Path = ""
+	}
+
+	// Assume "https" scheme if none is provided, and assume 8080 port for "http"
+	// scheme and 443 port for "https" scheme.
+	var secure bool
+	switch baseURLParsed.Scheme {
+	case "http":
+		baseURLParsed.Host = baseURLParsed.Host + ":" + "8080"
```

Comment:
If I supply something like `http://localhost:8080` does the host become `localhost:8080:8080` here? Or is that already handled elsewhere

Reply:
It was not handled; good catch! Added the missing `Port() == ""` check here.